### PR TITLE
Open window immediately on click for Facilities Booking

### PR DESCRIPTION
**Pull request by rldefa**

### CODE REVIEW

File: frontend/src/routes/FacilityBooking/ViewBooking/index.tsx [hunk 1]
```diff
@@ -145,17 +146,17 @@ export default function ViewBooking() {
...
     } catch (err) {
       console.log(err)
     }
   }
 
-  const openTelegram = (userID) => {
+  const openTelegram = (userID, tab) => {
     const site = 'https://telegram.me/' + userID
-    window.open(site)
+    if (!!tab) tab.location.href = site
```
Comment (hui444) [inline frontend/src/routes/FacilityBooking/ViewBooking/index.tsx:159]: why use `!!` ?

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
```
Comment (hui444) [inline yarn.lock:4]: why was this edited?

File: frontend/src/routes/FacilityBooking/ViewBooking/index.tsx [hunk 3]
```diff
@@ -145,17 +146,17 @@ export default function ViewBooking() {
           if (data.data === '' || data.data === undefined) {
             console.log(data.err)
           } else {
-            openTelegram(data.data)
+            openTelegram(data.data, newTab)
```
Comment (hui444) [inline frontend/src/routes/FacilityBooking/ViewBooking/index.tsx:149]: is this the only place using openTelegram?

File: yarn.lock (hunk 2 above)
Comment (rldefa) [reply yarn.lock:4]: generated by yarn, no idea

File: frontend/src/routes/FacilityBooking/ViewBooking/index.tsx (hunk 1 above)
Comment (rldefa) [reply frontend/src/routes/FacilityBooking/ViewBooking/index.tsx:159]: tab is nullable, so need check it isnt. !! converts to boolean

File: frontend/src/routes/FacilityBooking/ViewBooking/index.tsx (hunk 3 above)
Comment (rldefa) [reply frontend/src/routes/FacilityBooking/ViewBooking/index.tsx:149]: good catch, found one more and fixed in #8d98c31

File: yarn.lock (hunk 2 above)
Comment (rldefa) [reply yarn.lock]: yarn in wrong directory, fixed in 8d98c31

**Review by limwenfeng — APPROVED** [review]
LGTM!

**Pull request merged**